Packs/AWS-SNS-Listener/Integrations/AWSSNSListener/AWSSNSListener.py

```diff
@@ -0,0 +1,255 @@
+from CommonServerPython import *  # noqa: F401
+from CommonServerUserPython import *
+from tempfile import NamedTemporaryFile
+from traceback import format_exc
+from collections import deque
+import uvicorn
+from secrets import compare_digest
+from fastapi import Depends, FastAPI, Request, Response, status
+from fastapi.security import HTTPBasic, HTTPBasicCredentials
+from fastapi.security.api_key import APIKeyHeader
+from fastapi.openapi.models import APIKey
+import requests
+import base64
+from M2Crypto import X509
+
+
+PARAMS: dict = demisto.params()
+sample_events_to_store = deque(maxlen=20)  # type: ignore[var-annotated]
+
+app = FastAPI(docs_url=None, redoc_url=None, openapi_url=None)
+basic_auth = HTTPBasic(auto_error=False)
+token_auth = APIKeyHeader(auto_error=False, name='Authorization')
+
+
+def handle_teams_proxy_and_ssl():
+    proxies = None
+    use_ssl = not PARAMS.get('insecure', False)
+    if not is_demisto_version_ge('8.0.0'):
+        return proxies, use_ssl
+    CRTX_HTTP_PROXY = os.environ.get('CRTX_HTTP_PROXY', None)
+    demisto.debug(f"CRTX_HTTP_PROXY: {CRTX_HTTP_PROXY}")
+    if CRTX_HTTP_PROXY:
+        proxies = {
+            "http": CRTX_HTTP_PROXY,
+            "https": CRTX_HTTP_PROXY
+        }
+        use_ssl = True
+    return proxies, use_ssl
+
+
+PROXIES, USE_SSL = handle_teams_proxy_and_ssl()
+
+
+def valid_sns_message(sns_payload):
+    """
+    Validates an incoming Amazon Simple Notification Service (SNS) message.
+
+    Args:
+        sns_payload (dict): The SNS payload containing relevant fields.
+
+    Returns:
+        bool: True if the message is valid, False otherwise.
+    """
+    # taken from https://github.com/boto/boto3/issues/2508
+    # Can only be one of these types.
+    if sns_payload["Type"] not in ["SubscriptionConfirmation", "Notification", "UnsubscribeConfirmation"]:
+        demisto.error('Not a valid SNS message')
+        return False
+
+    # Amazon SNS currently supports signature version 1 or 2.
+    if sns_payload["SignatureVersion"] not in ["1", "2"]:
+        demisto.error('Not using the supported AWS-SNS SignatureVersion 1 or 2')
+        return False
```

Comment:
Please replace the [] with `.get()` to make the code more safe, also it'll be better to add the `Type` and `SignatureVersion` to the log message if condition is met.